subsys/net/lib/sockets/sockets.c

```diff
@@ -229,13 +315,40 @@ int zsock_accept(int sock, struct sockaddr *addr, socklen_t *addrlen)
 	return POINTER_TO_INT(ctx);
 }
 
-ssize_t zsock_send(int sock, const void *buf, size_t len, int flags)
+#ifdef CONFIG_USERSPACE
+Z_SYSCALL_HANDLER(zsock_accept, sock, addr, addrlen)
 {
-	return zsock_sendto(sock, buf, len, flags, NULL, 0);
+	socklen_t addrlen_copy;
+	int ret;
+
+	if (Z_SYSCALL_OBJ(sock, K_OBJ_NET_CONTEXT)) {
+		errno = EBADF;
+		return -1;
+	}
+
+	Z_OOPS(z_user_from_copy(&addrlen_copy, (void *)addrlen,
```

Comment:
OTOH, I don't understand why we allow kernel to write directly into addr, and don't allow into addrlen...

Reply:
addrlen is an input/output parameter whose contents must be copied to prevent TOCTOU issues.